File: pet_hospital_frontend/src/api/drug.ts
import api from './index'
import type { PageResult } from './customer'

export interface Drug {
    id?: number
    name: string
    code: string
    specification?: string
    manufacturer?: string
    price: number
    stockQuantity?: number
    warningThreshold?: number
    unit?: string
    isActive?: number
}

export const drugApi = {
    list: (params: { page: number; size: number; keyword?: string; status?: number }) => {
        return api.get<any, { code: number; data: PageResult<Drug> }>('/drugs', { params })
    },

    listActive: () =>
        api.get<any, { code: number; data: Drug[] }>('/drugs/active'),

    getLowStock: () =>
        api.get<any, { code: number; data: Drug[] }>('/drugs/low-stock'),

    getById: (id: number) =>
        api.get<any, { code: number; data: Drug }>(`/drugs/${id}`),

    create: (data: Drug) =>
        api.post<any, { code: number; data: boolean }>('/drugs', data),

    update: (id: number, data: Drug) =>
        api.put<any, { code: number; data: boolean }>(`/drugs/${id}`, data),

    delete: (id: number) =>
        api.delete<any, { code: number; data: boolean }>(`/drugs/${id}`)
}

File: pet_hospital_backend/src/main/java/com/pethospital/entity/Drug.java
package com.pethospital.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.math.BigDecimal;

@Data
@TableName("drugs")
public class Drug {
    @TableId(type = IdType.AUTO)
    private Long id;
    
    /**
     * 药品名称
     */
    private String name;
    
    /**
     * 药品编码
     */
    private String code;
    
    /**
     * 规格
     */
    private String specification;
    
    /**
     * 厂家
     */
    private String manufacturer;
    
    /**
     * 销售价格
     */
    private BigDecimal price;
    
    /**
     * 当前库存
     */
    private Integer stockQuantity;
    
    /**
     * 预警阈值
     */
    private Integer warningThreshold;
    
    /**
     * 单位: 盒, 瓶, 支
     */
    private String unit;
    
    /**
     * 是否启用
     */
    private Integer isActive;
}
